packages/api/src/blog.js

```diff
@@ -7,10 +8,19 @@ import { JSONResponse } from './utils/json-response.js'
  * @return {Promise<JSONResponse>}
  */
 export async function blogSubscriptionCreate (request, env, ctx) {
-  const requestData = await request.json()
-  const { email } = requestData
+  const prefix = env.MAILCHIMP_API_PREFIX
+  const listId = env.MAILCHIMP_BLOG_LIST_ID
+  const apiKey = env.MAILCHIMP_API_KEY
+  const headers = {
+    'Content-Type': 'application/json; charset=UTF-8',
+    Authorization: `Basic: ${apiKey}`
+  }
 
-  // placeholder for actual subscribe logic
-
-  return new JSONResponse({ email }, { status: 202 })
+  const body = await request.json();
+  (await isChimpUser(body.email, prefix, listId, headers))
```

Comment:
suggestion (non-blocking):
Personaly preference but there's a bit of complexity trying to grok this mentally, it might be better to just create meaningful variable names and well formed code paths instead of the 3 awaited functions + ternary

Reply:
@dashcraft agreed, explicit simplicity applied.